### DISCUSSION TITLE
Topdon TC-1 USB Thermal Camera

### DISCUSSION THREAD

**Discussion by wss7932** [TheRealWaldo/thermal#224] [Q&A]
I have a Topdon TC-1 USB-c camera and I currently have it hooked to my raspberry pi and am using Motion Eye, which simply sees it as a camera.  It works, but this app looks like what I really want.  I'd like to use this camera, which has good resolution, and to take advantage of its other capabilities such as temperature measurements.  Right now it just shows up as a "white-hot" thermal image.

**Comment by TheRealWaldo** [comment]
Having one makes it easier to design and use an application.  I don't have one.

That said, a few points:

 - Documentation about their protocols would be needed to determine any feasibility without reverse engineering (which would require access to the device).
 - It's possible this device doesn't send/make available anything other than compressed camera output, making it impossible to do accurate temperature measurements and next to impossible to get reliable data.
- Processing an encoded video stream rather than the sensor's raw output would be prohibitively CPU-intensive.

In short, for you to see any benefit, they'd need to expose the raw sensor data in a way that could be made available to my integration.

Do you have access to any of their protocol documentation?

**Reply by wss7932** [reply]
Sorry for going dark.  I recently came across this project which runs on a pi and uses the same camera. https://github.com/leswright1977/PyThermalCamera/  here is the video as well.  https://www.youtube.com/watch?v=PiVwZoQ8_jQ&t=44s

**Reply by TheRealWaldo** [reply]
Not sure I have the budget for one of these right now!
